react-js/react.md

```diff
@@ -47,53 +50,65 @@ You will spend the day working through these materials.
-4. Don't forget `this` when working within a React class. Examples: `this.state.thing`
-5. Don't forget that variables in JSX must be enclosed in brackets and don't forget number 4. Example: `<div>Hello {this.state.thing}</div>`
-6. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
-7. The term 'ReactJS' is commonly used to clarify that it's a javascript framework; however, 'React' is its official name.
+1. In render functions, don't forget to return JSX. You can return null for an empty output. 
+Remove the return from a component in one of the CodePens and take note of the error message shown.
+2. Don't forget that variables in JSX must be enclosed in brackets. Example: `<div>Hello {this.state.thing}</div>`
+3. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
 
 ### Guided Practice
-
--[Importing & Exporting with React](https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac)
+- [Rendering a List using map](https://codepen.io/jamesjose03/pen/jOPRwVm)
+  - View the [Codepen](https://codepen.io/jamesjose03/pen/jOPRwVm) which renders an  Ordered List using React. Try to modify the code and work along with it.
+  - Here, a class component named `FriendsContainer` is declared from the lines 1 - 19.
+  - Every class component has a render function which returns the code to be rendered.
+  - A few variables are declared such as name, friends, listItem. 
+  - `name` is just a string. `friends` is a list of Strings. `listItem` maps each element of the friends list as `li` tags.
+  - Inside return, the list is rendered by using `{ variableName }` this helps React understand that it has to render the value of the variable there. 
+  - `ReactDOM.render()` is used to render the component at the div which has the id `app` in the HTML file.
+  - Use ES6 arrow function
+  - Use semantic HTML, like `<ol>`
+  - If something is modified or updated, React needs to quickly figure out which element changed and where it needs to reflect that change. So, we have to set `key` attribute on the li of the list.
+    - Eg: `<li key={value}>`
+  - If the key property is not set, then the browser will give warning in the console.
 
 ### Independent Practice
+- [Rendering a List using map](https://codepen.io/jamesjose03/pen/jOPRwVm)
+- Try rendering the List by using declaring another component and passing the list to it for rendering it there.
+- Or try rendering the List in a different order by sorting it or so.
+
+- Other Requirements: 
+  - Modify the props or components.
+  - Play along with the code, modify it and see the changes it makes.
+  - Try rendering a basic calculator which does only the basic operations.
```

Comment:
Can you move this to a different list of alternative practice options? It's not really related to the other example. But Iike that you have other options in case the reader doesn't like the list example.

Reply:
Okay 👍